File: leetcode/Python/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sol,m,l,r = 0, collections.defaultdict(int),0,0
        for i in range(len(s)):
            if s[i] in m:
                l=max(l,m[s[i]]+1)
            m[s[i]], sol, r = i, max(sol,r-l+1), r+1
        return sol;

File: leetcode/C++/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int sol=0,l=0,r=0;
        unordered_map<char,int> um;
        for(int i=0; i<s.length(); ++i,++r) {
            if(um.find(s[i])!=um.end()) l=max(l,um[s[i]]+1);
            um[s[i]]=i;
            sol=max(sol, r-l+1);
        }
        return sol;
    }
};